tests/cdc-bank/bank.go

```diff
@@ -0,0 +1,249 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cdcbank
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	httputil "github.com/pingcap/tipocket/pkg/util/http"
+
+	"github.com/ngaut/log"
+
+	"github.com/pingcap/tipocket/pkg/cluster"
+	"github.com/pingcap/tipocket/pkg/core"
+	"github.com/pingcap/tipocket/util"
+)
+
+// Config contains the configurations of the CDC bank case
+type Config struct {
+	Accounts    int
+	Concurrency int
+	RunTime     time.Duration
+	Failpoints  map[string]string
+}
+
+// ClientCreator is a CDC bank creator
+type ClientCreator struct {
+	Cfg *Config
+}
+
+// Create creates a CDC bank client
+func (c ClientCreator) Create(_ cluster.ClientNode) core.Client {
+	return &client{
+		Config: c.Cfg,
+	}
+}
+
+const initBalance int = 1000
+const regions int = 16
+const validateConcurrency int = 32
+
+type client struct {
+	*Config
+}
+
+func pk(accountID int) int {
+	return accountID * 10000
+}
+
+func (c *client) SetUp(ctx context.Context, nodes []cluster.Node, clientNodes []cluster.ClientNode, idx int) error {
+	return nil
+}
+
+func (c *client) TearDown(ctx context.Context, nodes []cluster.ClientNode, idx int) error {
+	return nil
+}
+
+func (c *client) Invoke(ctx context.Context, node cluster.ClientNode, r interface{}) core.UnknownResponse {
+	panic("implement me")
+}
+
+func (c *client) NextRequest() interface{} {
+	panic("implement me")
+}
+
+func (c *client) DumpState(ctx context.Context) (interface{}, error) {
+	panic("implement me")
+}
+
+func (c *client) Start(ctx context.Context, cfg interface{}, clientNodes []cluster.ClientNode) error {
+	var upstream, downstream *sql.DB
+	expectedSum := c.Accounts * initBalance
+
+	for i, node := range clientNodes {
+		dsn := fmt.Sprintf("root@tcp(%s:%d)/test", node.IP, node.Port)
+		if i == 0 {
+			// upstream TiDB
+			log.Infof("set up upstream DB connection, dsn: %v", dsn)
+
+			// set failpoints
+			statusAddr := fmt.Sprintf("http://%s-tidb.%s.svc:10080", node.ClusterName, node.Namespace)
```

Comment:
I guess there may only one upstream TiDB server fail point is set because of the DNS round-robin?

PS I recommend you try this #352 if want to set the fail points of only the upstream cluster.